test/e2e/case14_selection_test.go

```diff
@@ -4,130 +4,266 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	"open-cluster-management.io/config-policy-controller/test/utils"
 )
 
 const (
-	case14PolicyNamed       string = "../resources/case14_namespaces/case14_limitrange_named.yaml"
-	case14PolicyUnnamed     string = "../resources/case14_namespaces/case14_limitrange_unnamed.yaml"
-	case14PolicyNamedName   string = "policy-named-limitrange"
-	case14PolicyUnnamedName string = "policy-unnamed-limitrange"
-	case14LimitRangeFile    string = "../resources/case14_namespaces/case14_limitrange.yaml"
-	case14LimitRangeName    string = "container-mem-limit-range"
+	case14LimitRangeFile string = "../resources/case14_namespaces/case14_limitrange.yaml"
+	case14LimitRangeName string = "container-mem-limit-range"
 )
 
-var _ = Describe("Test policy compliance with namespace selection", func() {
-	Describe("Create a named limitrange policy on managed cluster in ns: "+testNamespace, func() {
-		It("should be created properly on the managed cluster", func() {
-			By("Creating " + case14PolicyNamed + " on managed")
-			utils.Kubectl("apply", "-f", case14PolicyNamed, "-n", testNamespace)
+var _ = Describe("Test policy compliance with namespace selection", Ordered, func() {
+	checkRelated := func(policy *unstructured.Unstructured) []interface{} {
+		related, _, err := unstructured.NestedSlice(policy.Object, "status", "relatedObjects")
+		if err != nil {
+			panic(err)
+		}
+
+		return related
+	}
+
+	testNamespaces := []string{"range1", "range2"}
+	newNs := "range3"
+	policyTests := []struct {
+		name       string
+		yamlFile   string
+		hasObjName bool
+	}{
+		{
+			"policy-named-limitrange",
+			"../resources/case14_namespaces/case14_limitrange_named.yaml",
+			true,
+		},
+		{
+			"policy-unnamed-limitrange",
+			"../resources/case14_namespaces/case14_limitrange_unnamed.yaml",
+			false,
+		},
+	}
+
+	BeforeAll(func() {
+		By("Create Namespaces if needed")
+		namespaces := clientManaged.CoreV1().Namespaces()
+		for _, ns := range testNamespaces {
+			if _, err := namespaces.Get(context.TODO(), ns, metav1.GetOptions{}); err != nil && errors.IsNotFound(err) {
+				Expect(namespaces.Create(context.TODO(), &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: ns,
+					},
+				}, metav1.CreateOptions{})).NotTo(BeNil())
+			}
+			Expect(namespaces.Get(context.TODO(), ns, metav1.GetOptions{})).NotTo(BeNil())
+		}
+	})
+
+	AfterAll(func() {
+		for _, ns := range testNamespaces {
+			By("Deleting " + case14LimitRangeName + " on " + ns)
+			utils.Kubectl("delete", "-f", case14LimitRangeFile, "-n", ns)
+		}
+		for _, ns := range append(testNamespaces, newNs) {
+			By("Deleting namespace " + ns)
+			utils.Kubectl("delete", "namespace", ns, "--ignore-not-found")
+		}
+		for _, policy := range policyTests {
+			By("Deleting " + policy.name + " on managed")
+			utils.Kubectl("delete", "-f", policy.yamlFile, "-n", testNamespace)
```

Comment:
In some cases we would want to delete the policy first since it might recreate other resources that we want to delete.